packages/api/src/user/utils.ts

```diff
@@ -0,0 +1,28 @@
+import { UserAccountModel } from '@useraccount/entities';
+import { UserDocument } from './types';
+
+export const shortenEthAddress = (address: string): string => {
+  return `${address.substring(0, 6)}...${address.substring(
+    address.length - 4
+  )}`;
+};
+
+export const generateUserName = async (user: UserDocument): Promise<string> => {
```

Comment:
Nice, like we talked about, attempting to establish a single source of truth for every business rule. And to push it as far up the "tree" as possible with the database representing the "root node".